pkg/download/goget/goget.go

```diff
@@ -0,0 +1,161 @@
+package goget
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/gomods/athens/pkg/config"
+	"github.com/gomods/athens/pkg/config/env"
+	"github.com/gomods/athens/pkg/download"
+	"github.com/gomods/athens/pkg/errors"
+	"github.com/gomods/athens/pkg/module"
+	"github.com/gomods/athens/pkg/storage"
+	"github.com/spf13/afero"
+)
+
+// New returns a download protocol by using
+// go get. You must have a modules supported
+// go binary for this to work.
+func New() download.Protocol {
+	return &goget{
+		goBinPath: env.GoBinPath(),
+		fs:        afero.NewOsFs(),
+	}
+}
+
+type goget struct {
+	goBinPath string
+	fs        afero.Fs
+}
+
+func (gg *goget) List(ctx context.Context, mod string) ([]string, error) {
+	const op errors.Op = "goget.List"
+	lr, err := gg.list(op, mod)
+	if err != nil {
+		return nil, err
+	}
+
+	return lr.Versions, nil
+}
+
+type listResp struct {
+	Path     string
+	Version  string
+	Versions []string `json:"omitempty"`
+	Time     time.Time
+}
+
+func (gg *goget) Info(ctx context.Context, mod string, ver string) (*storage.RevInfo, error) {
+	const op errors.Op = "goget.Info"
+	fetcher, _ := module.NewGoGetFetcher(gg.goBinPath, gg.fs) // TODO: remove err from func call
+	ref, err := fetcher.Fetch(mod, ver)
+	if err != nil {
+		return nil, errors.E(op, err)
+	}
+	defer ref.Clear()
+	v, err := ref.Read()
+	if err != nil {
+		return nil, errors.E(op, err)
+	}
+	defer v.Zip.Close()
```

Comment:
It's kind of the same thing but you could probably just call Close as the zip is never used.